src/test/java/com/google/cloud/tools/managedcloudsdk/NullProgressListener.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2018 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.managedcloudsdk;
+
+public class NullProgressListener implements ProgressListener {
```

Comment:
It might simplify the other implementers to provide an incremental `ProgressListener` that pushes up updates as they come in.  I think something like the following should work (untested), taking in to account the `UNKNOWN` idea earlier.
```java
interface ProgressListener {
  /** Special value for situations where the total work cannot be determined in advance. */
  long UNKNOWN = -1;

  // etc.
}
class SubProgressListener implements ProgressListener {
  private ProgressListener parent;
  private long parentTotalWork;
  private long parentReported = 0; // accumulated
  private long totalWork;
  private long reported = 0; // accumulated

  public SubProgressListener(ProgressListener parent, long totalWork) {
    Preconditions.checkArgument(totalWork > 0);
    this.parent = parent;
    this.parentTotalWork = totalWork;
  }

  public void start(String message, long totalWork) {
    Preconditions.checkArgument(totalWork > 0 || totalWork == UNKNOWN)
    this.totalWork = totalWork;
  }

  public void update(long worked) {
    Preconditions.checkArgument(worked > 0);
    reported += worked;
    if(totalWork > 0) {
      Preconditions.checkState(reported <= totalWork);
      // ensure not affected by rounding (e.g., parentTotalWork == 1)
      long parentWorked = reported * parentTotalWork / totalWork - parentReported;
      if(parentWorked > 0) {
        parentReported += parentWorked;
        parent.update(parentWorked);
      }
    }
  }

  public void done() {
    if(parentReported < parentTotalWork) {
      update(parentTotalWork - parentReported);
    }
  }
}
```

Reply:
@briandealwis does this handle the `unknown` case with exponential backoff or were you saying that might be an option?

Reply:
Never mind I just reread what you said, that eclipse provides the implementation for -1.

Reply:
I'll add an implementation in a follow up PR.